src/contracts/pods/EigenPodManager.sol

```diff
@@ -267,8 +259,42 @@ contract EigenPodManager is
         return (currentDepositShares < 0 ? 0 : uint256(currentDepositShares), shares);
     }
 
+    /// @dev Calculates the proportion a pod owner's restaked balance has decreased, and
+    /// reduces their beacon slashing factor accordingly.
+    /// Note: `balanceDecreasedWei` is assumed to be less than `prevRestakedBalanceWei`
+    function _reduceSlashingFactor(
+        address podOwner,
+        uint256 prevRestakedBalanceWei,
+        uint256 balanceDecreasedWei
+    ) internal returns (uint256, uint64, uint256) {
+        // Apply negative balance delta to calculate the proportion of the original
+        // balance that remains. Note that underflow here should be impossible given
+        // the invariants pods use to calculate these values.
+        uint256 newRestakedBalanceWei = prevRestakedBalanceWei - balanceDecreasedWei;
+        uint256 balanceRemainingWad = newRestakedBalanceWei.divWad(prevRestakedBalanceWei);
+
+        // Update pod owner's beacon chain slashing factor. Note that `newBeaconSlashingFactor`
+        // should be less than `prevBeaconSlashingFactor` because `balanceRemainingWad` is
+        // guaranteed to be less than WAD.
+        uint64 prevBeaconSlashingFactor = beaconChainSlashingFactor(podOwner);
+        uint64 newBeaconSlashingFactor = uint64(prevBeaconSlashingFactor.mulWad(balanceRemainingWad));
+        emit BeaconChainSlashingFactorDecreased(podOwner, newBeaconSlashingFactor);
+        /// forgefmt: disable-next-item
+        _beaconChainSlashingFactor[podOwner] = BeaconChainSlashingFactor({
+            slashingFactor: newBeaconSlashingFactor,
+            isSet: true
+        });
+
+        uint256 curDepositShares = uint256(podOwnerDepositShares[podOwner]);
+        // Note that underflow here should be impossible given
+        // `balanceRemainingWad` is guaranteed to be less than WAD.
+        uint256 wadSlashed = uint256(WAD) - balanceRemainingWad;
```

Comment:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/901
Already made the relevant change in this PR but wanted to note I think we should round down the amount slashed by changing the `divWad` in calculating `balanceRemainingWad` to be rounded upwards.
We do this similarly in `SlashingLib.calcSlasheAmount` specifically for underflow edge cases

Reply:
[3d4e2a6](https://github.com/Layr-Labs/eigenlayer-contracts/pull/892/commits/3d4e2a6d10792ebb36c436610ddbe0113084d373)